### ISSUE TITLE
How to use @register_job for multiple tasks? There some mistakes or errors, Please tell me it's true or false!

### ISSUE THREAD

**Issue by zychen2016** [jcass77/django-apscheduler#110]
**Describe the bug / feature**

Anconda:django-apscheduler        0.3.1

I have write a project with django, Now There are five tasks about update mysql databases 。 I want to update in some interval。for example update one time every 3 hours。

However，There some fields in my logs，I couldn't understand. 
 

```
2020-09-14 01:02:30,373-add_item-INFO-项目添加总共耗时: 2.728264331817627
2020-09-14 01:02:30,373-add_item-INFO-项目添加总共耗时: 2.728264331817627
2020-09-14 01:02:30,797-update_filter_sample-INFO-过滤数据更新总共耗时：2.893929958343506
2020-09-14 01:02:30,797-update_filter_sample-INFO-过滤数据更新总共耗时：2.893929958343506
2020-09-14 01:04:28,798-off_machine_date-INFO-更新新下机时间总共耗时: 120.69730019569397
2020-09-14 01:04:28,798-off_machine_date-INFO-更新新下机时间总共耗时: 120.69730019569397
2020-09-14 01:06:54,777-update_sample-INFO-样品添加总共耗时:267.0045871734619
2020-09-14 01:06:54,777-update_sample-INFO-样品添加总共耗时:267.0045871734619
...
...
...
2020-09-14 04:02:31,482-add_item-INFO-项目添加总共耗时: 2.6115708351135254
2020-09-14 04:02:31,482-add_item-INFO-项目添加总共耗时: 2.6115708351135254
2020-09-14 04:02:31,482-add_item-INFO-项目添加总共耗时: 2.6115708351135254
2020-09-14 04:02:32,154-update_filter_sample-INFO-过滤数据更新总共耗时：2.9415314197540283
2020-09-14 04:02:32,154-update_filter_sample-INFO-过滤数据更新总共耗时：2.9415314197540283
2020-09-14 04:02:32,154-update_filter_sample-INFO-过滤数据更新总共耗时：2.9415314197540283
2020-09-14 04:04:36,666-off_machine_date-INFO-更新新下机时间总共耗时: 127.23617815971375
2020-09-14 04:04:36,666-off_machine_date-INFO-更新新下机时间总共耗时: 127.23617815971375
2020-09-14 04:04:36,666-off_machine_date-INFO-更新新下机时间总共耗时: 127.23617815971375
2020-09-14 04:07:07,900-update_sample-INFO-样品添加总共耗时:278.8854019641876
2020-09-14 04:07:07,900-update_sample-INFO-样品添加总共耗时:278.8854019641876
2020-09-14 04:07:07,900-update_sample-INFO-样品添加总共耗时:278.8854019641876

```

Why it print two rows about 01:02 , But print 3 rows about 04:02? one more time? Is this normal?

**To Reproduce**

`views.py`: Five functions about update mysql databases. (1):add_data (2):update_sample (3):update_filter_sample (4):update_off_machine (5):update_item_flag. (1)~(4) update every 3 hours , (5) update every 8 hours.

```
try:
    # 实例化调度器
    scheduler = BackgroundScheduler()
    scheduler.add_jobstore(DjangoJobStore(), "default")


    @register_job(scheduler, "interval", hours=3, misfire_grace_time=1000, coalesce=True)
    def add_data():
        """
        每六小时添加一次项目
        :return:
        """
        from .addItem import get_item_ndays
        import time
        from utils.MyLogger import MyLogger

        my_log = MyLogger("add_item").get_log()
        start_time = time.time()
        # 增加2天前立项的Item
        get_item_ndays(days=60)
        end_time = time.time()
        my_log.info(f"项目添加总共耗时: {end_time - start_time}")


    @register_job(scheduler, "interval", hours=3, misfire_grace_time=1000, coalesce=True)
    def update_sample():
        """
        添加此刻24小时前添加的项目
        :return:
        """
        from django.db.models import F, Q
        from sample_table.add_sample import update_or_create_sample
        from .models import Item
        import datetime
        import time
        from utils.MyLogger import MyLogger

        my_log = MyLogger("update_sample").get_log()
        start_time = time.time()
        items = Item.objects.filter(item_flag__lt=3)
        for item in items:
            update_or_create_sample(item.sub_code)
        end_time = time.time()
        my_log.info(f"样品添加总共耗时:{end_time - start_time}")


    @register_job(scheduler, "interval", hours=3, misfire_grace_time=1000, coalesce=True)
    def update_filter_sample():
        """
        更新项目的已过滤数
        :return:
        """
        from .models import Item
        from sample_table.models import SampleInfo
        import datetime
        import time
        from utils.MySingleton import RedisCache
        from utils.MyLogger import MyLogger

        my_log = MyLogger("update_filter_sample").get_log()
        rd = RedisCache()
        start_time = time.time()
        items = Item.objects.filter(item_flag__lt=3)
        for item in items:
            result = rd.get_data(f'filter_{item.sub_code}')['result']
            times = rd.get_data(f'filter_time_{item.sub_code}')['result']
            filter_count = SampleInfo.objects.filter(sub_code=item.sub_code, is_filter=True).count()
            # 首次设置或者0 时
            if not result and not times:
                rd.set_data(f'filter_{item.sub_code}', filter_count, 90 * 24 * 60 * 60)
                rd.set_data(f'filter_time_{item.sub_code}', 1, 90 * 24 * 60 * 60)
                item.is_filter_percent = filter_count
                item.save()
            # 变化更新,切更新次数
            elif item.is_filter_percent != int(result) and int(times) < 4:
                times = int(times) + 1
                rd.set_data(f'filter_{item.sub_code}', filter_count)
                rd.set_data(f'filter_time_{item.sub_code}', times)
                item.is_filter_percent = filter_count
                item.save()

        end_time = time.time()
        my_log.info(f"过滤数据更新总共耗时：{end_time - start_time}")


    @register_job(scheduler, "interval", hours=3, misfire_grace_time=1000, coalesce=True)
    def update_off_machine():
        """
        更新下机时间
        :return:
        """
        from .models import Item
        from .addItem import get_off_machine_date
        from utils.MySingleton import RedisCache

        import datetime
        import time
        from django.db.models import F, Q
        from utils.MyLogger import MyLogger

        my_log = MyLogger("off_machine_date").get_log()
        rd = RedisCache()
        start_time = time.time()

        items = Item.objects.filter(item_flag__lt=3)

        for item in items:
            off_machine_date = get_off_machine_date(item.sub_code)
            times = rd.get_data(f'off_time_{item.sub_code}')['result']
            result = rd.get_data(f'off_{item.sub_code}')['result']
            # 首次
            if not result:
                rd.set_data(f'off_{item.sub_code}', str(off_machine_date), 90 * 24 * 60 * 60)
                rd.set_data(f'off_time_{item.sub_code}', 0, 90 * 24 * 60 * 60)
                item.off_machine_date = off_machine_date
                item.save()
            elif str(off_machine_date) != str(item.off_machine_date) and int(times) < 3:
                times = int(times) + 1
                rd.set_data(f'off_{item.sub_code}', off_machine_date)
                rd.set_data(f'off_time_{item.sub_code}', times)
                item.off_machine_date = off_machine_date
                item.save()

        end_time = time.time()
        my_log.info(f"更新新下机时间总共耗时: {end_time - start_time}")


    @register_job(scheduler, "interval", hours=8, misfire_grace_time=1000, coalesce=True)
    def update_item_flag():
        """
        更新项目状态和邮件提醒功能
        """
        from item_table.models import Item
        from item_table.addItem import get_item_flag, get_item_unfiltered
        from utils.MySingleton import RedisCache

        items = Item.objects.filter(item_flag__lt=3)
        rd = RedisCache()
        for item in items:
            error_time = rd.get_data(f'error_{item.sub_code}')['result']
            if error_time and int(error_time) > 3:
                continue
            get_item_flag(item.sub_code)
        get_item_unfiltered()


    register_events(scheduler)
    scheduler.start()
    print("Scheduler started!")

except Exception as e:
    print(e)
    scheduler.shutdown()

```

And, why the `update_item_flag` do not run in 8 hours? It seems that it never run .

How to use `@register_job` for multiple tasks?  There are any mistakes about my code?

**Expected behavior**

It should only print one time when update .

**Screenshots**

**Comment by jcass77** [comment]
Please refer to the [caveats](https://github.com/jarekwg/django-apscheduler#caveats) section in the README.

Using `BackgroundScheduler` in Django (where multiple web server processes can be running in production deployments) is inherently unsafe and will require that you implement [your own remote processing logic](https://apscheduler.readthedocs.io/en/latest/faq.html#how-do-i-share-a-single-job-store-among-one-or-more-worker-processes) to regulate access to one shared job store.

There have also been a large number of changes and fixes since the v0.3.1 release and you may want to check out the most recent release of django-apscheduler first.

**Comment by zychen2016** [comment]
> Please refer to the [caveats](https://github.com/jarekwg/django-apscheduler#caveats) section in the README.
> 
> Using `BackgroundScheduler` in Django (where multiple web server processes can be running in production deployments) is inherently unsafe and will require that you implement [your own remote processing logic](https://apscheduler.readthedocs.io/en/latest/faq.html#how-do-i-share-a-single-job-store-among-one-or-more-worker-processes) to regulate access to one shared job store.
> 
> There have also been a large number of changes and fixes since the v0.3.1 release and you may want to check out the most recent release of django-apscheduler first.

Thank you，I will have a try。
